php/src/Google/Protobuf/Internal/FileOptions.php

```diff
@@ -549,6 +554,34 @@ public function hasPyGenericServices()
     }
 
     /**
+     * Generated from protobuf field <code>optional bool php_generic_services = 19 [default = false];</code>
+     * @return bool
+     */
+    public function getPhpGenericServices()
+    {
+        return $this->php_generic_services;
+    }
+
+    /**
+     * Generated from protobuf field <code>optional bool php_generic_services = 19 [default = false];</code>
+     * @param bool $var
+     * @return $this
+     */
+    public function setPhpGenericServices($var)
+    {
+        GPBUtil::checkBool($var);
+        $this->php_generic_services = $var;
+        $this->has_php_generic_services = true;
+
+        return $this;
+    }
+
+    public function hasPhpGenericServices()
+    {
+        return $this->has_php_generic_services;
```

Comment:
The generated code for descriptor.proto is currently for implementation usage only.
descriptor.proto is a proto2 file instead of proto3. That's why it has the hasX method.